packages/wallet/test/integration/txChainingBalance.test.ts

```diff
@@ -0,0 +1,32 @@
+import { Cardano } from '@cardano-sdk/core';
+import { SingleAddressWallet } from '../../src';
+import { createWallet } from './util';
+import { firstValueFrom } from 'rxjs';
+
+describe('integration/txChainingBalance', () => {
+  let wallet: SingleAddressWallet;
+
+  beforeAll(async () => {
+    wallet = await createWallet();
```

Comment:
Maybe add a comment here to ensure it's clear we're using a mocked tx submit provider

Reply:
How would another provider make it produce a false positive? :thinking: If you're thinking `wallet.balance` being updated from the provider then I don't think it could ever happen, even if the transaction got on-chain instantly upon submission and the wallet polling request matched the same tick, because it's only polling the `tip`, balance (utxo) request triggered after that

Reply:
Right, it wouldn't. The comment is just to communicate the behavior is isolated from any network transmission. I also immediately wondered which provider was being used, so it would clarify just this point alone.

Reply:
Added a few comments in 4c94a3ff807c435c5aa27a04b46037fee3495d22

Reply:
Further improved the test by asserting precondition in ccb6fa7aa7ac6b8c0dc4f6a23d12ef370a78a5d2